### ISSUE TITLE
SpatialAverager having issues in ESMF 8.2.0

### ISSUE THREAD

**Issue by aulemahal** [pangeo-data/xESMF#131]
Build failures of #122 are currently due to the change to ESMF 8.2.0.

There seems to be a change in the Mesh object that makes it return a wrong shape (as compared to 8.1.1 and to the list of polygons).  It then fails in an unrelated line further down the line.

MWE:
```python3
import xesmf as xe
from shapely import Polygon

# Those are the two last polygons of test_frontend.py::polys
p1 = Polygon([[3, 3], [3, 4], [4, 4], [4, 3]])
p2 = Polygon([[0, 0], [0.5, 0.5], [0, 1], [0.5, 1.5], [0, 2], [2, 2], [1.5, 1.5], [2, 1], [1.5, 0.5], [2, 0]])

xe.backend.Mesh.from_polygons([p1])
# Mesh:
#     rank = 1
#     size = [4, 1]
#     size_owned = [4, 1]
#     coords = [[array([3., 4., 4., 3.]), array([3., 3., 4., 4.])], [array([3.5]), array([3.5])]]

xe.backend.Mesh.from_polygons([p2])
# Mesh:
#     rank = 1
#     size = [10, 8]  <-  should be [10, 1]
#     size_owned = [10, 1]
#     coords = [[array([0. , 2. , 1.5, 2. , 1.5, 2. , 0. , 0.5, 0. , 0.5]), array([0. , 0. , 0.5, 1. , 1.5, 2. , 2. , 1.5, 1. , 0.5])], [array([1.]), array([1.])]]
```
On ESMF 8.1.1 the size of the second mesh is [10, 1], which is correct. On ESMF 8.2.0, there is this error that comes out of nowhere. I'm investigating.

@huard You said you had issues with 8.2.0 ? Do they relate to this one?

**Comment by huard** [comment]
I think so, yes.

I had to do
```w = _parse_coords_and_values(weights, self.n_in, mesh_out.element_count)```
instead of
```w = _parse_coords_and_values(weights, self.n_in, self.n_out)```
to get it to work.

**Comment by huard** [comment]
A fix is an #128, but I didn't dig deep.

**Comment by aulemahal** [comment]
Ryan O'Kuinghttons of NOAA responded to my email raising the issue. He is working on fixing it as it really is a bug.

> The internal Mesh class in ESMF stores ngons as triangles. The output for Mesh.size you are now seeing in xESMF is the count of the number of triangles that is used to represent the ngons that were originally defined. We have historically endeavored to only return the information that was originally specified for the Mesh, but it looks like some of the internal ESMF information was leaked to the user interface.

Moreover, #128 implicitly fixed this issue because the `size` attribute of the Mesh (which is wrong) is never used explicitly anymore.

@huard In your case, `self.n_out` was not correct anyway since the number of elements in the mesh doesn't correspond to the number of geometries in the regridder! In the first call of `_parse_coords_and_values` the number of element includes exploded MultiPolygons and in the second call, it's only holes. In both cases, not `n_out`.
